drivers/clock_control_nrfxlib/clock_control_nrfxlib.c

```diff
@@ -168,22 +193,65 @@ void nrf_power_clock_isr(void)
 		power_event_cb(NRF_POWER_EVENT_USBREMOVED);
 	}
 #endif
+
+#ifdef CONFIG_CLOCK_CONTROL_BT_LL_NRFXLIB
 	/* FIXME/NOTE: This handler is called _after_ the USB registers are
 	 * checked. The reason is that the BLE controller also checks these
 	 * registsers _and clears them_, but there is currently no event
 	 * propagated from the controller.
 	 */
 	ble_controller_POWER_CLOCK_IRQHandler();
+#endif
 }
 
+
+#if !defined(CONFIG_CLOCK_CONTROL_BT_LL_NRFXLIB)
+void m_assert_handler(const char *const file, const u32_t line)
+{
+	/*  TODO: Improve fault handler.*/
+	k_oops();
+}
+#endif
+
 static int clock_control_init(struct device *dev)
 {
+#ifdef CONFIG_CLOCK_CONTROL_BT_LL_NRFXLIB
 	ARG_UNUSED(dev);
-
 	/* No-op. Initialized in subsys/bluetooth/controller/hci_driver.c by
 	 * ble_init() at PRE_KERNEL_1. ble_init() will call
 	 * ble_controller_init() that in turn will start the LFCLK.
 	 */
+#else /* CONFIG_CLOCK_CONTROL_MPSL_NRFXLIB */
+	__ASSERT_NO_MSG(dev != NULL);
+
+	if (dev->config->name == "CLOCK_16M") {
+
+	} else if (dev->config->name == "CLOCK_32K") {
+		int err = 0;
+		mpsl_clock_lf_cfg_t clock_cfg;
+
+#ifdef CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC
+		clock_cfg.source = MPSL_CLOCK_LF_SRC_RC;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_XTAL
+		clock_cfg.source = MPSL_CLOCK_LF_SRC_XTAL;
+#elif CLOCK_CONTROL_NRF_K32SRC_SYNTH
+		clock_cfg.source = MPSL_CLOCK_LF_SRC_SYNTH;
+#else
+#error "Clock source is not defined"
+#endif
+
+		clock_cfg.rc_ctiv = MPSL_RECOMMENDED_RC_CTIV;
+		clock_cfg.rc_temp_ctiv = MPSL_RECOMMENDED_RC_TEMP_CTIV;
+
+		err = mpsl_init(&clock_cfg, SWI5_IRQn, m_assert_handler);
```

Comment:
Try to make an MPSL subsys (fw-nrfconnect-nrf) where we initialize MPSL. Like hci_driver initializes ble_controller.

Reply:
Done. Seems to work.